suite-common/wallet-core/src/accounts/accountsActions.ts

```diff
@@ -0,0 +1,137 @@
+                // xrp `availableBalance` is reduced by reserve, use regular balance
+                discoveryItem.networkType === 'ripple'
+                    ? accountInfo.balance
+                    : accountInfo.availableBalance,
+                discoveryItem.coin,
+            ),
+            tokens: enhanceTokens(accountInfo.tokens),
+            addresses: enhanceAddresses(
+                accountInfo.addresses,
+                discoveryItem.networkType,
+                discoveryItem.index,
+            ),
+            utxo: enhanceUtxo(accountInfo.utxo, discoveryItem.networkType, discoveryItem.index),
+            history: accountInfo.history,
+            metadata: {
+                key: accountInfo.legacyXpub || accountInfo.descriptor,
+                fileName: '',
+                aesKey: '',
+                outputLabels: {},
+                addressLabels: {},
+            },
+            ...getAccountSpecific(accountInfo, discoveryItem.networkType),
+        },
+    }),
+);
+
+const updateAccount = createAction(
+    `${actionPrefix}/updateAccount`,
+    (account: Account, accountInfo: AccountInfo | null = null): { payload: Account } => {
+        if (accountInfo) {
+            return {
+                payload: {
+                    ...account,
+                    ...accountInfo,
+                    path: account.path,
+                    empty: accountInfo.empty,
+                    visible: account.visible || !accountInfo.empty,
+                    formattedBalance: formatNetworkAmount(
+                        // xrp `availableBalance` is reduced by reserve, use regular balance
+                        account.networkType === 'ripple'
+                            ? accountInfo.balance
+                            : accountInfo.availableBalance,
+                        account.symbol,
+                    ),
+                    utxo: enhanceUtxo(accountInfo.utxo, account.networkType, account.index),
+                    addresses: enhanceAddresses(
+                        accountInfo.addresses,
+                        account.networkType,
+                        account.index,
+                    ),
+                    tokens: enhanceTokens(accountInfo.tokens),
+                    ...getAccountSpecific(accountInfo, account.networkType),
+                },
+            };
+        }
+        return {
+            payload: account,
+        };
+    },
+);
+
+const changeAccountVisibility = createAction(
+    `${actionPrefix}/changeAccountVisibility`,
+    (account: Account, visible = true): { payload: Account } => ({
+        payload: {
+            ...account,
+            visible,
+        },
+    }),
+);
+
+export const accountsActions = {
+    disposeAccount,
+    removeAccount,
+    createAccount,
+    updateAccount,
+    updateSelectedAccount,
+    changeAccountVisibility,
+} as const;
+
+export type AccountsAction = ReturnType<typeof accountsActions[keyof typeof accountsActions]>;
```

Comment:
Please move this type directly to packages/suite where it is used, this action union types are bad and we need it only for legacy reason so I don't want to encourage anyone to use if kept here.